components/toast_wrapper/toast_wrapper.tsx

```diff
@@ -2,103 +2,95 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import PropTypes from 'prop-types';
-import {FormattedMessage, injectIntl} from 'react-intl';
+import {FormattedMessage, injectIntl, WrappedComponentProps} from 'react-intl';
+import {RouteComponentProps} from 'react-router-dom';
+
+import {Preferences} from 'mattermost-redux/constants';
 
-import Toast from 'components/toast/toast';
-import Timestamp, {RelativeRanges} from 'components/timestamp';
 import {isIdNotPost, getNewMessageIndex} from 'utils/post_utils';
-import * as Utils from 'utils/utils';
+import {isKeyPressed, localizeMessage} from 'utils/utils';
 import {isToday} from 'utils/datetime';
 import Constants from 'utils/constants';
 import {browserHistory} from 'utils/browser_history';
+
+import Toast from 'components/toast/toast';
+import Timestamp, {RelativeRanges} from 'components/timestamp';
 import {SearchShortcut} from 'components/search_shortcut';
 import {HintToast} from 'components/hint-toast/hint_toast';
-import {Preferences} from 'mattermost-redux/constants';
+
+import type {OwnProps, PropsFromRedux} from '.';
 
 const TOAST_TEXT_COLLAPSE_WIDTH = 500;
 
 const TOAST_REL_RANGES = [
     RelativeRanges.TODAY_YESTERDAY,
 ];
 
-class ToastWrapper extends React.PureComponent {
-    static propTypes = {
-        unreadCountInChannel: PropTypes.number,
-        newRecentMessagesCount: PropTypes.number,
-        channelMarkedAsUnread: PropTypes.bool,
-        isCollapsedThreadsEnabled: PropTypes.bool,
-        rootPosts: PropTypes.object,
-        atLatestPost: PropTypes.bool,
-        postListIds: PropTypes.array,
-        latestPostTimeStamp: PropTypes.number,
-        atBottom: PropTypes.bool,
-        lastViewedBottom: PropTypes.number,
-        width: PropTypes.number,
-        lastViewedAt: PropTypes.number,
-        focusedPostId: PropTypes.string,
-        initScrollOffsetFromBottom: PropTypes.number,
-        updateNewMessagesAtInChannel: PropTypes.func,
-        scrollToNewMessage: PropTypes.func,
-        scrollToLatestMessages: PropTypes.func,
-        scrollToUnreadMessages: PropTypes.func,
-        updateLastViewedBottomAt: PropTypes.func,
-        showSearchHintToast: PropTypes.bool,
-        onSearchHintDismiss: PropTypes.func,
-        shouldStartFromBottomWhenUnread: PropTypes.bool,
-        isNewMessageLineReached: PropTypes.bool,
-        unreadScrollPosition: PropTypes.string,
-
-        /*
-         * Object from react-router
-         */
-        match: PropTypes.shape({
-            params: PropTypes.shape({
-                team: PropTypes.string,
-            }).isRequired,
-        }).isRequired,
-
-        actions: PropTypes.shape({
-
-            /**
-             * Action creator to update toast status
-             */
-            updateToastStatus: PropTypes.func.isRequired,
-        }).isRequired,
-    };
-
+export type Props = PropsFromRedux & OwnProps & WrappedComponentProps & RouteComponentProps<{team: string}> & {
```

Comment:
You don't need to import and extend the PropsFromRedux and ownProps. It's more readable to declare all properties in this object and not need to import back from the index file, as this causes circular imports. You can still declare them as interfaces and use in the index.ts itself

```
interface OwnProps {
 ....
}
```

```function mapStateToProps(state: GlobalState, ownProps: OwnProps) {```

Reply:
Recently, I started to use `PropsFromRedux` in mattermost-webapp repo, because I was seeing that it is being used by staff and if I remember correctly, @hmhealey was suggesting to use it. Anyhow, it is not a big difference to me. I changed it as suggested 🚀